CI Failing - Linux py37_np_dev - test_constructor_list_frames

appeared on latest push to master, and recently updated PRs.

```
=================================== FAILURES ===================================
____________ TestDataFrameConstructors.test_constructor_list_frames ____________
[gw1] linux -- Python 3.7.3 /home/vsts/miniconda3/envs/pandas-dev/bin/python

self = <pandas.tests.frame.test_constructors.TestDataFrameConstructors object at 0x7fd3adeb3278>

    def test_constructor_list_frames(self):
        # see gh-3243
        result = DataFrame([DataFrame()])
>       assert result.shape == (1, 0)
E       assert (1, 1) == (1, 0)
E         At index 1 diff: 1 != 0
E         Use -v to get the full diff

pandas/tests/frame/test_constructors.py:119: AssertionError
```

--- a/pandas/tests/frame/test_constructors.py
+++ b/pandas/tests/frame/test_constructors.py
@@ -15,7 +15,7 @@ from pandas.core.dtypes.common import is_integer_dtype
 import pandas as pd
 from pandas import (
     Categorical, DataFrame, Index, MultiIndex, RangeIndex, Series, Timedelta,
-    Timestamp, compat, date_range, isna)
+    Timestamp, date_range, isna)
 from pandas.tests.frame.common import TestData
 import pandas.util.testing as tm
 
@@ -113,7 +113,6 @@ class TestDataFrameConstructors(TestData):
         assert df.loc[1, 0] is None
         assert df.loc[0, 1] == '2'
 
-    @pytest.mark.xfail(compat.numpy._is_numpy_dev, reason="GH-26546")
     def test_constructor_list_frames(self):
         # see gh-3243
         result = DataFrame([DataFrame()])

